### DISCUSSION TITLE
using SHAP explainer in a FLASK app

### DISCUSSION THREAD

**Discussion by MahsaShokouhi** [shap/shap#1876] [Q&A]
Hi,
I've created a prediction model app using FLASK and would like to use SHAP to explain my model prediction. I'm wondering how to integrate SHAP explainer into FLASK app. More specifically, I'd like to convert either SHAP explainer (output of shap.KernelExplainer()) or shap force_plot to html format in real-time (without saving the plot for example). Any suggestions would be very much appreciated.
Thanks in advance,

**Comment by MohsenAskar** [comment]
Hi @MahsaShokouhi :) , any luck finding an answer for this? I have the same issue.

**Comment by MahsaShokouhi** [comment]
Yes @MohsenAskar . The shap plot has a html() method which you can use. For example, let's say you have a data frame df, and an explainer:

shap_values = explainer.shap_values(df)
f_plot = shap.force_plot(explainer.expected_value, shap_values, df)

then using f_plot.html() inside the render_template method of Flask would render the plot in the html file.

**Reply by MohsenAskar** [reply]
Thanks, @MahsaShokouhi !! 🙏
